huaweicloud/services/rds/resource_huaweicloud_rds_pg_hba.go

```diff
@@ -0,0 +1,263 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product RDS
+// ---------------------------------------------------------------
+
+package rds
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// API: RDS POST /v3/{project_id}/instances/{instance_id}/hba-info
+// API: RDS GET /v3/{project_id}/instances/{instance_id}/hba-info
+func ResourcePgHba() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePgHbaCreateOrUpdate,
+		UpdateContext: resourcePgHbaCreateOrUpdate,
+		ReadContext:   resourcePgHbaRead,
+		DeleteContext: resourcePgHbaDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the RDS PostgreSQL instance.`,
+			},
+			"host_based_authentications": {
+				Type:        schema.TypeList,
+				Elem:        pgHbaHostBasedAuthenticationSchema(),
+				Required:    true,
+				MinItems:    1,
```

Comment:
the **Required** attr already means that the MinItems is 1, so I think it's unnecessary to add this line.